### ISSUE TITLE
Usage of default buffer for xxx

### ISSUE THREAD

**Issue by clement-brodu** [Riverside-Software/sonar-openedge#1065]
## My versions : 
- OpenEdge plugin version: 2.21.0-98e8b7ed
- OE 12.7.0
- Sonarlint 7.6.2312

## My case :

I'm using the static Regex class from c#, that make a false positive of `Usage of default buffer for PwRegD`

```openedge
BLOCK-LEVEL ON ERROR UNDO, THROW.

USING System.Text.RegularExpressions.Regex FROM ASSEMBLY.

DEFINE VARIABLE vChar AS CHARACTER NO-UNDO.

vChar = Regex:Escape("Hello"). //false positive 'Usage of default buffer for PwRegD'

MESSAGE 
   vChar
   VIEW-AS ALERT-BOX INFORMATION BUTTONS OK IN WINDOW CURRENT-WINDOW.
```

I have this field in my database : 

```openedge
ADD FIELD "REGExt" OF "PwRegD" AS integer 
  FORMAT "9999"
  INITIAL "0"
  LABEL "Extent"
  POSITION 5
  MAX-WIDTH 4
  ORDER 50
```

See full table in this DF :  
[df.zip](https://github.com/Riverside-Software/sonar-openedge/files/12332817/df.zip)

**Comment by gquerret** [comment]
More a proparse issue than a BufferUsage rule issue.

**Comment by gquerret** [comment]
Using an even more simplified version of your code:
```
using System.Text.RegularExpressions.Regex from assembly.
message Regex:Escape("test").
```

This code compiles if your table is not in the schema, but does **not** compile if your table is in the schema.
OE 12.2

**Comment by gquerret** [comment]
Same with OE 12.7. Unless there's another case where the BufferUsage rule is triggered on valid code, that wont' be fixed.

**Comment by clement-brodu** [comment]
```abl
using System.Text.RegularExpressions.Regex from assembly.
message Regex:Escape("test").
```

This code compile if you set `COMPILER:OPTIONS = "require-full-names,require-field-qualifiers,require-return-values"`

**Comment by gquerret** [comment]
Note: Block.lookupField() should have an option to mimic the behavior of require* options.

**Comment by gquerret** [comment]
New boolean property `sonar.oe.require.full_names` added in the develop branch. Note that this doesn't fail the generation of the JPNode tree, but no symbol will be attached to the RecordNameNode object.
Default behavior is unchanged ; testing is required on various codebases to see the impact on rules.
